logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultIterableLikeAssertions.kt

```diff
@@ -117,6 +117,24 @@ class DefaultIterableLikeAssertions : IterableLikeAssertions {
         converter: (T) -> Iterable<E>
     ): FeatureExtractorBuilder.ExecutionStep<T, E> = collect(container, converter, "max", Iterable<E>::max)
 
+    override fun <T : Any, E> containsNoDuplicates(
+        container: AssertionContainer<T>,
+        converter: (T) -> Iterable<E>
+    ): Assertion {
+        val finalList = mutableListOf<E>()
```

Comment:
Since you asked for more hints, something like the following (copied from IterableLikeAssertions.all - check the output if this is actually what we want (I did not check it myself). Check the necessary information you want to see in the spec
```
val iterableContainer = container.changeSubject.unreported(converter).toAssertionContainer()
        return LazyThreadUnsafeAssertionGroup {
            val list = turnSubjectToList(iterableContainer).maybeSubject.getOrElse { emptyList() }
            val hasElementAssertion = createHasElementAssertion(list.iterator())

            val duplicates: List<Pair<Int, E>> = TODO("determine duplicates including their index - see createMismatchAssertions for an idea")

            val duplicateAssertions = duplicates.map {
                //see createMismatchAssertions: extract the creation of the descriptive assertion and reuse here as well
            }
            assertionBuilder.invisibleGroup
                .withAssertions(
                    hasElementAssertion,
                    assertionBuilder.fixedClaimGroup
                        .withListType
                        .withClaim(duplicates.isEmpty())
                        .withDescriptionAndRepresentation(DescriptionBasic.HAS_NOT, DescriptionIterableAssertion.DUPLICATE_ELEMENTS)
                        .withAssertions(duplicateAssertions)
                        .build()
                )
                .build()
        }
```

Reply:
very good hint :D, implement it